wv2util/ExperimentalFeature.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace wv2util
+{
+    public class ExperimentalFeature : IEquatable<ExperimentalFeature>, IComparable<ExperimentalFeature>
+    {
+        public string Name { get; set; }
+        
+        public bool isEnabled;
+
+        public bool IsEnabled
+        {
+            get { return isEnabled; }
+            set
+            {
+                if (value)
+                {
+                    turnOn();
+                }
+                else
+                {
+                    turnOff();
+                }
+                isEnabled = value;
+            }
+        }
+
+        public string Description { get; set; }
+
+        private Action turnOn, turnOff;
+
+        public ExperimentalFeature(Action turnOn, Action turnOff, Func<bool> isOn)
+        {
+            this.turnOn = turnOn;
+            this.turnOff = turnOff;
+            isEnabled = isOn();
+        }
+
+        static public bool IsEnabledForApp(string commandLine, string envVars)
+        {
+            // TODO implement
+            return false;
+        }
+
+        public int CompareTo(ExperimentalFeature other)
+        {
+            return Name.CompareTo(other.Name);
+        }
+
+        public bool Equals(ExperimentalFeature other)
+        {
+            return Name.Equals(other.Name);
+        }
+    }
+
+    public class EnvVarExperimentalFeature : ExperimentalFeature
+    {
+        public EnvVarExperimentalFeature(string envVar, string onVal, string offVal) : base(
+            () =>
+            {
+                // Turn on:
+                Environment.SetEnvironmentVariable(envVar, onVal, EnvironmentVariableTarget.User);
+            },
+            () =>
+            {
+                // Turn off:
+                Environment.SetEnvironmentVariable(envVar, offVal, EnvironmentVariableTarget.User);
+            },
+            () =>
+            {
+                string val = Environment.GetEnvironmentVariable(envVar, EnvironmentVariableTarget.User);
+                return val != null && val == onVal;
+            })
+        {
+        }
+    }
+
+    public class ExperimentalFeatureList : ObservableCollection<ExperimentalFeature>
+    {
+        private List<ExperimentalFeature> m_List = new List<ExperimentalFeature>();
+
+        public ExperimentalFeatureList()
+        {
+            // Visual Hosting
+            Items.Add(new EnvVarExperimentalFeature(
+                "COREWEBVIEW2_FORCED_HOSTING_MODE",
+                "COREWEBVIEW2_HOSTING_MODE_WINDOW_TO_VISUAL",
+                "COREWEBVIEW2_HOSTING_MODE_WINDOW_TO_WINDOW")
```

Comment:
Shouldn't off for this be the lack of the environment variable? Otherwise you'll start with no env var, check will add the env var with value 1, uncheck will set the env var to value 2 and there's no way to go back to no env var. If value 2 and no env var are equivalent let's normalize to no env var.

Reply:
That's correct, I just checked with Brad.